docs/src/tutorials/manifold.md

```diff
@@ -0,0 +1,218 @@
+# [How to implement your own manifold](@id manifold-tutorial)
+
+```@meta
+CurrentModule = ManifoldsBase
+DocTestSetup  = quote
+    using Manifolds
+end
+```
+
+This tutorial demonstrates how to easily set your own manifold up within `Manifolds.jl`.
+
+## Introduction
+
+If you looked around a little and saw the [interface](../interface.md), the amount of functions and possibilities, it might seem that a manifold might take some time to implement.
+This tutorial demonstrates that you can get your first own manifold quite fast and you only have to implement the functions you actually need. For this tutorial it would be helpful if you take a look at our [notation](../notation.md).
+This tutorial assumes that you heard of the exponential map, tangent vectors and the dimension of a manifold. If not, please read for example [[do Carmo, 1992](#doCarmo1992)],
+Chapter 3, first.
+
+We start with two technical preliminaries. If you want to start directly, you can [skip](@ref manifold-tutorial-task) this paragraph and revisit it for two of the implementation details.
+
+After that, we will
+
+* [model](@ref manifold-tutorial-task) the manifold
+* [implement](@ref manifold-tutorial-checks) two tests, so that points and tangent vectors can be checked for validity, for example also within [`ValidationManifold`](@ref),
+* [implement](@ref manifold-tutorial-fn) two functions, the exponential map and the manifold dimension.
+
+## [Technical preliminaries](@id manifold-tutorial-prel)
+
+There are only two small technical things we need to explain at this point.
+First of all our [`Manifold`](@ref)`{𝔽}` has a parameter `𝔽`.
+This parameter indicates the [`number_system`](@ref) the manifold is based on, for example `ℝ` for real manifolds. It is important primarily for defining bases of tangent spaces.
+See [`Symmetric`](@ref) as an example of defining both a real-valued and a complex-valued symmetric manifolds using one type.
+
+Second, a main design decision of `Manifold.jl` is that most functions are implemented as mutating functions, i.e. as in-place-computations. There usually exists a non-mutating version that falls back to allocating memory and calling the mutating one. This means you only have to implement the mutating version, _unless_ there is a good reason to provide a special case for the non-mutating one, i.e. because in that case you know a far better performing implementation.
+
+Let's look at an example. The exponential map $\exp_p\colon T_p\mathcal M \to \mathcal M$ that maps a tangent vector $X\in T_p\mathcal M$ from the tangent space at $p\in \mathcal M$ to the manifold.
```

Comment:
```suggestion
Let's look at an example. The exponential map $\exp_p\colon T_p\mathcal M \to \mathcal M$ that maps a tangent vector $X\in T_p\mathcal M$ from the tangent space at $p\in \mathcal M$ to the manifold. You don't have to implement it for your manifold but it is very useful.
```

Reply:
What is the aim of this new sentence? It's anyways meant as an example, so I don't feel its necessary to state that it's a useful function (and sometimes you even can't implement it – in closed form).

Reply:
Primarily I wouldn't want to leave the reader with an impression that the exponential map is necessary. Maybe it's poorly worded but at the moment it's somewhat unclear which functions are necessary for which functionality. Although that may be an idea for a different part of documentation. What do you think?

Reply:
You're right, that's a good point, but I'd like to add a paragraph in the introduction then; will think about one. I think here, also the place is not best, since its in the technical details section.

Reply:
Good idea, a paragraph in the introduction would be better 👍 .